python/pyspark/sql/connect/client/core.py

```diff
@@ -1143,6 +1148,26 @@ def execute_command_as_iterator(
                     },
                 )
 
+    def execute_ml(self, req: pb2.ExecutePlanRequest) -> Optional[pb2.MlCommandResponse]:
+        """
+        Execute the ML command request and return ML response result
+        Parameters
+        ----------
+        req : pb2.ExecutePlanRequest
+            Proto representation of the plan.
+        """
+        logger.info("Execute ML")
+        try:
+            for attempt in self._retrying():
+                with attempt:
+                    for b in self._stub.ExecutePlan(req, metadata=self._builder.metadata()):
+                        if b.HasField("ml_command_result"):
+                            return b.ml_command_result
```

Comment:
This is a bit tricky here as you don't fully consume the result iterator. In addition, this needs to be integrated with the progress handler that we have. 

Wouldn't it be better to use the https://github.com/apache/spark/blob/master/python/pyspark/sql/connect/client/core.py#L1095 method?

Reply:
Or basically we should model it similarly.